forem/forem
# Get a11y lint script paths from rails helper

## What type of PR is this? (check all applicable)


## Description

The accessibility linting feature introduced in #13745 currently isn't working in non-Heroku Forems. This is because the JS is looking for an asset path which doesn't include the fingerprinted asset name, and it's therefore not being resolved or loaded in.

Rails helpers allow us to get the accurate asset path, and as these are needed in the Preact code, I've added them as data attributes on a hidden `div` that we can then retrieve from the Preact code. This is a similar pattern to what we do in various places where data needs passed from Rails to the Javascript code, and I've added a comment in the JS to explain why we're doing this.

## Related Tickets & Documents

#13745

## QA Instructions, Screenshots, Recordings

To test this, we need to check that the accessibility linter is functional in both the "Edit post" and "New post" views. Ideally we'd test this in an environment that relied on accessing fingerprinted versions of the assets, but I might need advice on how to do that!

https://user-images.githubusercontent.com/20773163/120770014-9b92b400-c515-11eb-8ffc-6a76f8c21dfc.mp4


### UI accessibility concerns?

N/A - this only changes how the script is loaded in

## Added tests?

- [X] No, and this is why: the current tests test the feature

## [Forem core team only] How will this change be communicated?

_Will this PR introduce a change that impacts Forem members or creators, the
development process, or any of our internal teams? If so, please note how you
will share this change with the people who need to know about it._

      [Admin Guide](https://forem.gitbook.io/forem-admin-guide/), or
      [Storybook](https://storybook.forem.com/) (for Crayons components)
      or in a [forem.dev](http://forem.dev) post
- [X] This change does not need to be communicated, and this is why not: we haven't communicated the original feature yet

## [optional] Are there any post deployment tasks we need to perform?

We may need to re-check this functionality in the fleet, to make sure it's definitely working
